submitit/helpers.py

```diff
@@ -216,3 +218,55 @@ def __enter__(self) -> None:
 
     def __exit__(self, *args):
         os.chdir(self.original_dir)
+
+
+def monitor_jobs(
+    jobs: tp.Sequence[core.Job[core.R]], sleep_time_s: float = 30, test_mode: bool = False
+) -> None:
+    """Continuously monitors given jobs until they are all done or failed.
+
+    Parameters
+    ----------
+    jobs: List[Jobs]
+        A list of jobs to monitor
+    sleep_time_s: int
+        The time (in seconds) between two refreshes of the monitoring.
+        Can't be inferior to 30s.
+    test_mode: bool
+        If in test mode, we do not check the length of sleep_time_s
+    """
+
+    if not test_mode:
+        assert sleep_time_s >= 30, "You can't refresh too often (>= 30s) to avoid overloading squeue"
+
+    monitoring_start_time = time.time()
+    n_jobs = len(jobs)
+    while True:
+        state_jobs = collections.defaultdict(set)
+        for i, job in enumerate(jobs):
+            if job.state.upper() in ["FAILED", "RUNNING"]:
+                state_jobs[job.state.upper()].add(i)
+            elif job.done():
+                state_jobs["DONE"].add(i)
```

Comment:
I'm not clear about why this condition is needed, if it's not failed, running, or pending, then it's completed (or completing for a small amount of time). So this seems unecessary complexity. Any reason then?